pkg/operator/configobserver/apiserver/observe_tlssecurityprofile.go

```diff
@@ -100,10 +100,17 @@ func getSecurityProfileCiphers(profile *configv1.TLSSecurityProfile) (string, []
 	}
 
 	// nothing found / custom type set but no actual custom spec
+	defaultProfileSpec := configv1.TLSProfiles[configv1.TLSProfileIntermediateType]
 	if profileSpec == nil {
-		profileSpec = configv1.TLSProfiles[configv1.TLSProfileIntermediateType]
+		profileSpec = defaultProfileSpec
+	}
+
+	// If a custom spec was provided with no MinTLSVersion, use the default
+	minTLSVersion := profileSpec.MinTLSVersion
+	if minTLSVersion == "" {
+		minTLSVersion = defaultProfileSpec.MinTLSVersion
```

Comment:
this changes the value (behaviour) for existing resources, right ? Do you know which TLS version the api servers would use previously ? (customProfile, no minTLSVersion set). which other components use the TLS profile ?

Reply:
Yes, this does change the observed value for other components when no minTLSVersion is set. [KCM for example](https://github.com/openshift/cluster-kube-controller-manager-operator/pull/506/files#diff-0af15f182aea9898e784fc589175ef4c2411c9391bb5a1f41327ba7c3e2172f5) omits the --tls-min-version argument if minTLSVersion is empty. After this change, the observer will no longer return an empty string, which means KCM will explicitly set --tls-min-version to the value returned by the observer.

If omitting --tls-min-version completely is a better option, I can try to rework how storage components respond to this instead of changing the observer. But then we're deferring the decision of "reasonable default" to each component when the argument is omitted instead of making the decision here. This means different components could assume different default values.